airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java

```diff
@@ -164,12 +162,11 @@ public void acceptTracked(final AirbyteMessage message) throws InterruptedExcept
   private void processRecord(final AirbyteMessage message) throws InterruptedException {
     final var streamId = AirbyteStreamNameNamespacePair.fromRecordMessage(message.getRecord());
     uploaderMap.get(streamId).upload(message);
-
-    // This is just modular arithmetic written in a complicated way. We want to run T+D every
-    // RECORDS_PER_TYPING_AND_DEDUPING_BATCH records.
-    // TODO this counter should be per stream, not global.
-    if (recordsSinceLastTDRun.getAndUpdate(l -> (l + 1) % RECORDS_PER_TYPING_AND_DEDUPING_BATCH) == RECORDS_PER_TYPING_AND_DEDUPING_BATCH - 1) {
```

Comment:
I don't know if getting the current time in Java is slow or not... but there's probably some overheard. We might want to keep the modulo above, and only do this check every N records to save some compute time.

Reply:
Did some testing and grabbing the system time for each record does add a bit of overhead, updated the `TypeAndDedupeOperationValve` to add a record count check to only check the timing every 100 records